.circleci/config.yml

```diff
@@ -31,24 +42,68 @@ deployment:
+      - checkout
+      - run: git submodule update --init
+      # Install dependencies.
+      - run: brew install awscli cmake
       # Build the MacOS demo app and package it into demo.zip
-      - make osx -j 2 MACOSX_DEPLOYMENT_TARGET=10.10.0
-      - cd build/osx/bin && zip -r ${CIRCLE_ARTIFACTS}/demo.zip tangram.app
+      - run: make osx -j 2 MACOSX_DEPLOYMENT_TARGET=10.10.0
+      - run: cd build/osx/bin && zip -r ${CIRCLE_ARTIFACTS}/demo.zip tangram.app
       # Upload the demo archive to S3.
-      - aws s3 cp ${CIRCLE_ARTIFACTS}/demo.zip s3://ios.mapzen.com/tangram-osx-snapshots/tangram-snapshot-debug-${CIRCLE_BUILD_NUM}.zip
-      - aws s3 cp ${CIRCLE_ARTIFACTS}/demo.zip s3://ios.mapzen.com/tangram-osx-latest-debug.zip
-  releases:
-    # For any tag of the form 1, 1.2.3, 1.4-beta2, etc. we will deploy a release build.
-    tag: /[0-9]+(\.[0-9]+)*(-beta[0-9]*)?/
-    owner: tangrams
-    commands:
+      - run: aws s3 cp ${CIRCLE_ARTIFACTS}/demo.zip s3://ios.mapzen.com/tangram-osx-snapshots/tangram-snapshot-debug-${CIRCLE_BUILD_NUM}.zip
+      - run: aws s3 cp ${CIRCLE_ARTIFACTS}/demo.zip s3://ios.mapzen.com/tangram-osx-latest-debug.zip
+  build-deploy-ios-release:
+    macos:
+      xcode: "9.0"
+    environment:
+      # Dummy values, Circle won't run without a project and scheme.
+      XCODE_PROJECT: build/tangram.xcodeproj
+      XCODE_SCHEME: phony
+    steps:
+      # Check out repository with submodules.
+      - checkout
+      - run: git submodule update --init
+      # Install dependencies.
+      - run: brew install awscli cmake
+      - run: gem install jazzy
       # Build the framework in release mode and package it into pod.zip
-      - make clean-ios
-      - make ios-framework-universal RELEASE=1
-      - cd build/ios-framework-universal/Release/ && zip -r ${CIRCLE_ARTIFACTS}/pod.zip TangramMap.framework
+      - run: make ios-framework-universal RELEASE=1
+      - run: make ios-docs
+      - run: cd build/ios-framework-universal/Release/ && zip -r ${CIRCLE_ARTIFACTS}/pod.zip TangramMap.framework
       # Add the readme file.
-      - cd platforms/ios/framework/ && zip ${CIRCLE_ARTIFACTS}/pod.zip README.md
+      - run: cd platforms/ios/framework/ && zip ${CIRCLE_ARTIFACTS}/pod.zip README.md
       # Add the license file.
-      - zip ${CIRCLE_ARTIFACTS}/pod.zip LICENSE
+      - run: zip ${CIRCLE_ARTIFACTS}/pod.zip LICENSE
       # Upload the Cocoapods archive to S3.
-      - aws s3 cp ${CIRCLE_ARTIFACTS}/pod.zip s3://ios.mapzen.com/tangram-releases/tangram-release-${CIRCLE_TAG}.zip
+      - run: aws s3 cp ${CIRCLE_ARTIFACTS}/pod.zip s3://ios.mapzen.com/tangram-releases/tangram-release-${CIRCLE_TAG}.zip
+
+workflows:
+  version: 2
+  build-and-deploy:
+    jobs:
+      - build-ios:
+          filters:
+            branches:
+              ignore: master
+      - build-deploy-macos-snapshot:
+          filters:
+            branches:
+              only: master
+      - build-deploy-ios-snapshot:
```

Comment:
`workflow` feature in the newer circle version is nice. I think based on the recommendation in https://circleci.com/docs/2.0/migrating-from-1-2/, we should split `build-universal-debug` and `deploy-snapshot`. As mentioned in the docs, this will prevent us to not run the entire build when just the deployment failed.

>`It might make sense to start by just splitting out a deploy job to prevent you from having to re-run the entire build when only the deployment fails.`

Reply:
Similar split the `build-deploy-ios-release` workflow to `build-universal-release` and `deploy-release`.

Reply:
For the above to work, we will need to use the `requires` key on the workflow jobs' `build` and `deploy` to run sequentially.